numpy/core/tests/test_multiarray.py

```diff
@@ -1073,6 +1073,9 @@ def test_zero_width_string(self):
         xx = x['S'].reshape((2, 2))
         assert_equal(xx.itemsize, 0)
         assert_equal(xx, [[b'', b''], [b'', b'']])
+        # check for no uninitialized memory due to viewing S0 array
+        repr_str = ''.join(repr(xx).replace('array', 'np.array').splitlines())
+        assert_array_equal(eval(repr_str), xx)
```

Comment:
Are you sure ravel makes a copy? I've found a bunch of places with zeros-sized items that it returns a view, and generally does the right thing.

Either way, the indexing tests the right thing, so no changes needed.